### PR TITLE
Dependency Update: web3-provider-engine@14.1.0

### PR DESCRIPTION
**Pull request by CruzMolina**
Take `@truffle/hdwallet-provider` off the `trufflesuite` hacky fork of `web3-provider-engine@~14.0.6` and onto `web3-provider-engine@14.1.0`.

Also unleashes a necessary shim to get `Web3` working with `web3-provider-engine`.
`web3-provider-engine` no longer supports synchronous requests, so we shim Web3 & also tweak `HdWalletProvider.send` to use `engine.sendAsync`.

QA/Smoke tested locally.

### CODE REVIEW

File: packages/hdwallet-provider/src/index.ts [hunk 1]
```diff
@@ -177,7 +177,7 @@ class HDWalletProvider {
     payload: JSONRPCRequestPayload,
     callback: JSONRPCErrorCallback | Callback<JsonRpcResponse>
   ): void {
-    return this.engine.send.call(this.engine, payload, callback);
+    return this.engine.sendAsync.call(this.engine, payload, callback);
```
Comment (gnidan) [inline packages/hdwallet-provider/src/index.ts]: really?

Comment (CruzMolina) [reply packages/hdwallet-provider/src/index.ts]: Ya, have an alternative in mind?

Comment (eggplantzzz) [reply packages/hdwallet-provider/src/index.ts]: What do you mean by really? @CruzMolina can you explain exactly what this is about?

Comment (CruzMolina) [reply packages/hdwallet-provider/src/index.ts]: Need to look into this a bit more. Web3 might not have actually dropped `sendAsync`.

The problem is Web3 `1.x` came out and formally removed support for `sendAsync`. `web3-provider-engine` responded by removing support for sync requests & forcing providers to have a `sendAsync` method.

Comment (CruzMolina) [reply packages/hdwallet-provider/src/index.ts]: Idk, the options here kind've suck.

We could downgrade Web3 here to `0.x` & not have to do the hack on this line (or the shimming).

**Comment by CruzMolina** [comment]
Closing since there doesn't seem to be interest in making the necessary adjustments to upgrade this dependency and transition `@truffle/hdwallet-provider` off of our fork.

**Pull request closed without merging**